staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go

```diff
@@ -106,9 +105,24 @@ func SerializeObject(mediaType string, encoder runtime.Encoder, hw http.Response
 		statusCode:      statusCode,
 		contentEncoding: negotiateContentEncoding(req),
 		hw:              hw,
-		ctx:             ctx,
 	}
 
+	defer func() {
+		if w.totalBytes == 0 {
+			return
+		}
+
+		var rErr = "<nil>"
+		if w.lastErr != nil {
+			rErr = w.lastErr.Error()
+		}
+
+		span.AddEvent("ResponseWriter",
```

Comment:
Can you preserve those events ""About to start writing response" and "Write call failed/succeeded" instead of changing the names totally.